clients/src/main/java/org/apache/kafka/common/requests/MetadataRequest.java

```diff
@@ -25,25 +25,40 @@
 import java.util.List;
 
 public class MetadataRequest extends AbstractRequest {
-    
+
     private static final Schema CURRENT_SCHEMA = ProtoUtils.currentRequestSchema(ApiKeys.METADATA.id);
     private static final String TOPICS_KEY_NAME = "topics";
 
     private final List<String> topics;
 
+    /**
+     * Constructor to support asking for no topic metadata
+     */
+    public MetadataRequest() {
```

Comment:
@ijuma I didn't realize compatibility wasn't a concern on these classes. Then we can do whatever we would like here.